core/libraries/shortcodes/EE_Primary_Registration_Details_Shortcodes.lib.php

```diff
@@ -184,13 +184,31 @@ protected function _parser($shortcode)
             }
 
             foreach ($primary_registration->questions as $ansid => $question) {
-                if (trim($question->get('QST_display_text')) == trim($shortcode)
+                if ($question instanceof EE_Question
+                    && trim($question->get('QST_display_text')) == trim($shortcode)
                     && isset($primary_registration->registrations[ $primary_reg->ID() ]['ans_objs'][ $ansid ])
                 ) {
-                    return $primary_registration->registrations[ $primary_reg->ID() ]['ans_objs'][ $ansid ]->get_pretty(
-                        'ANS_value',
-                        'no_wpautop'
-                    );
+                    $primary_reg_ansid = $primary_registration->registrations[ $primary_reg->ID() ]['ans_objs'][ $ansid ];
+                    
+                    // what we show for the answer depends on the question type!
+                    switch ($question->get('QST_type')) {
+                        case 'STATE':
```

Comment:
should use the constant representations for these types.  So for STATE it would be `EEM_Question::QST_type_state` for `COUNTRY` it would be `EEM_Question::QST_type_country`